clang/lib/Driver/Driver.cpp

```diff
@@ -979,6 +1000,45 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
     return;
   }
 
+  //
+  // SYCL
+  //
+  // We need to generate a SYCL toolchain if the user specified -fsycl.
+  // If -fsycl is supplied without any of these we will assume SPIR-V.
+  bool HasValidSYCLRuntime =
+      C.getInputArgs().hasFlag(options::OPT_fsycl, options::OPT_fno_sycl,
+                               false);
+
+  auto argSYCLIncompatible = [&](OptSpecifier OptId) {
+    if (!HasValidSYCLRuntime)
+      return;
+    if (Arg *IncompatArg = C.getInputArgs().getLastArg(OptId))
+      Diag(clang::diag::err_drv_argument_not_allowed_with)
+          << IncompatArg->getSpelling() << "-fsycl";
+  };
+  // -static-libstdc++ is not compatible with -fsycl.
+  argSYCLIncompatible(options::OPT_static_libstdcxx);
+  // -ffreestanding cannot be used with -fsycl
+  argSYCLIncompatible(options::OPT_ffreestanding);
+
+  llvm::SmallVector<llvm::Triple, 4> UniqueSYCLTriplesVec;
+
+  // If -fsycl is supplied without -fsycl-targets we will assume SPIR-V.
+  // For -fsycl-device-only, we also setup the implied triple as needed.
+  if (HasValidSYCLRuntime) {
+    addSYCLDefaultTriple(C, UniqueSYCLTriplesVec);
+  }
+
+  // We'll need to use the SYCL and host triples as the key into
```

Comment:
Do we need to guard this somehow so that it gets called for SYCL compilation? I see IsCuda and IsHIP being computed for those offloading models. Please take a look at this check and see if we need something similar.
https://github.com/mdtoguchi/llvm-project/blob/d819f8911ad6360b667289bd90eb5b6401105335/clang/lib/Driver/Driver.cpp#L812

Reply:
I can add a guard.